bonsai-core/src/main/scala/com/stripe/bonsai/FullBinaryTree.scala

```diff
@@ -59,14 +64,13 @@ object FullBinaryTree {
    *
    * @param tree the tree whose structure we are copying
    */
-  def apply[T](tree: T)(implicit ev: FullBinaryTreeOps.WithLayout[T]): FullBinaryTree[ev.treeOps.BranchLabel, ev.treeOps.LeafLabel] = {
+  def apply[T, B, L](tree: T)(implicit ev: FullBinaryTreeOps[T, B, L]): FullBinaryTree[B, L] = {
```

Comment:
That would be great. And, yeah, agree that `Aux` and `WithLayout` only make things confusing! I'm really happy they're gone :D